src/views/ModelPlan/TaskList/Beneficiaries/Frequency/index.tsx

```diff
@@ -362,28 +369,75 @@ const Frequency = () => {
-                          {beneficiariesT('precedenceRules.sublabel')}
-                        </p>
-
-                        <FieldErrorMsg>
-                          {flatErrors.precedenceRules}
-                        </FieldErrorMsg>
-
-                        <Field
-                          as={TextAreaField}
-                          className="height-15"
-                          error={flatErrors.precedenceRules}
-                          id="beneficiaries-precedence-rules"
-                          data-testid="beneficiaries-precedence-rules"
+                        <FieldArray
                           name="precedenceRules"
+                          render={arrayHelpers => (
+                            <>
+                              <legend className="usa-label">
+                                {beneficiariesT('precedenceRules.label')}
+                              </legend>
+
+                              <p className="text-base margin-top-1 margin-bottom-0 line-height-body-3">
+                                {beneficiariesT('precedenceRules.sublabel')}
+                              </p>
+
+                              <FieldErrorMsg>
+                                {flatErrors.precedenceRules}
+                              </FieldErrorMsg>
+
+                              {getKeys(beneficiaryPrecedenceConfig.options).map(
+                                key => (
+                                  <Fragment key={key}>
+                                    <Field
+                                      as={CheckboxField}
+                                      id={`beneficiaries-precedence-rules-${key}`}
+                                      data-testid={`beneficiaries-precedence-rules-${key}`}
+                                      name={`precedence-rules-${key}`}
+                                      label={
+                                        beneficiaryPrecedenceConfig.options[key]
+                                      }
+                                      value={key}
+                                      checked={values.precedenceRules.includes(
+                                        key
+                                      )}
+                                      onChange={(
+                                        e: React.ChangeEvent<HTMLInputElement>
+                                      ) => {
+                                        if (e.target.checked) {
+                                          arrayHelpers.push(e.target.value);
+                                        } else {
+                                          const idx = values.precedenceRules.indexOf(
+                                            e.target.value as YesNoType
+                                          );
+                                          arrayHelpers.remove(idx);
+                                        }
+                                      }}
+                                    />
```

Comment:
You shouldn't need the `FieldArray component, not the onChange handler here.  The CheckboxField should be able to handle this automatically.